Lib/typing.py

```diff
@@ -299,51 +300,22 @@ def __deepcopy__(self, memo):
         return self
 
 
-class _SpecialForm(_Final, _Immutable, _root=True):
-    """Internal indicator of special typing constructs.
-    See _doc instance attribute for specific docs.
-    """
-
-    __slots__ = ('_name', '_doc')
-
-    def __new__(cls, *args, **kwds):
-        """Constructor.
-
-        This only exists to give a better error message in case
-        someone tries to subclass a special typing object (not a good idea).
-        """
-        if (len(args) == 3 and
-                isinstance(args[0], str) and
-                isinstance(args[1], tuple)):
-            # Close enough.
-            raise TypeError(f"Cannot subclass {cls!r}")
-        return super().__new__(cls)
-
-    def __init__(self, name, doc):
-        self._name = name
-        self._doc = doc
-
-    @property
-    def __doc__(self):
-        return self._doc
-
-    def __eq__(self, other):
-        if not isinstance(other, _SpecialForm):
-            return NotImplemented
-        return self._name == other._name
+# Internal indicator of special typing constructs.
+# See __doc__ instance attribute for specific docs.
+class _SpecialForm(_Final, _root=True):
+    __slots__ = ('_name', '__doc__', '_getitem')
 
-    def __hash__(self):
-        return hash((self._name,))
+    def __init__(self, getitem):
+        self._getitem = getitem
+        self._name = getitem.__name__
+        self.__doc__ = getitem.__doc__
 
     def __repr__(self):
         return 'typing.' + self._name
 
     def __reduce__(self):
         return self._name
 
-    def __call__(self, *args, **kwds):
-        raise TypeError(f"Cannot instantiate {self!r}")
```

Comment:
Again, a custom exception message may be helpful, but not important.

Reply:
Your are right. "`Cannot instantiate typing.Any`" looks better than "`'_SpecialForm' object is not callable`". Initially I renamed `_SpecialForm` to `special form`, but later reverted this change.

Reply:
Perfect! Thanks for fixing this.